# HG changeset patch
# User Rik <rik@octave.org>
# Date 1378939460 25200
#      Wed Sep 11 15:44:20 2013 -0700
# Node ID f58ad514372ccf7df0a6adf8e5f2e5279ef199ae
# Parent  f47cfca56eb980c892bad2c79532198876ce8bf1
__contour__.m: Turn off auto clim adjustment while patches are being added.
And then turn back on when complete.  Performance increase.

* scripts/plot/private/__contour__.m: Set climmode to manual while adding
patch objects and then restore mode at end of for loop.

diff --git a/scripts/plot/private/__contour__.m b/scripts/plot/private/__contour__.m
--- a/scripts/plot/private/__contour__.m
+++ b/scripts/plot/private/__contour__.m
@@ -230,16 +230,20 @@ function add_patch_children (hg)
   zlev = get (hg, "zlevel");
   zmode = get (hg, "zlevelmode");
   lc = get (hg, "linecolor");
   lw = get (hg, "linewidth");
   ls = get (hg, "linestyle");
   filled = get (hg, "fill");
   ca = gca ();
 
+  ## Turn off automatic updating of clim while adding patches
+  climmode = get (ca, "climmode");
+  set (ca, "climmode", "manual"); 
+
   if (strcmpi (lc, "auto"))
     lc = "flat";
   endif
 
   if (strcmpi (filled, "on"))
 
     lvl_eps = get_lvl_eps (lev);
 
@@ -388,16 +392,18 @@ function add_patch_children (hg)
                                "facecolor", "none", "edgecolor", lc,
                                "linestyle", ls, "linewidth", lw,
                                "cdata", clev, "parent", hg)];
       endswitch
       i1 += clen + 1;
     endwhile
   endif
 
+  set (ca, "climmode", climmode);
+
 endfunction
 
 function update_zlevel (h, d)
 
   z = get (h, "zlevel");
   zmode = get (h, "zlevelmode");
   kids = get (h, "children");
 
